### PR TITLE
Gh756_SeriesImportDao_setSeriesIdAndChangeStatus_replace_arguments_by_dto_object

### PR DESCRIPTION
**Pull request by KrivenkoAlexander**
Addressed to #756

### CODE REVIEW

File: src/main/java/ru/mystamps/web/feature/series/importing/JdbcSeriesImportDao.java [hunk 1]
```diff
@@ -97,26 +96,21 @@ public Integer add(ImportSeriesDbDto importRequest) {
 	
 	// @todo #735 SeriesImportDao.setSeriesIdAndChangeStatus(): replace arguments by dto object
```
Comment (php-coder) [inline src/main/java/ru/mystamps/web/feature/series/importing/JdbcSeriesImportDao.java]: We need to remove this comment altogether.

File: src/test/groovy/ru/mystamps/web/feature/series/importing/SeriesImportServiceImplTest.groovy [hunk 2]
```diff
@@ -258,14 +258,14 @@ class SeriesImportServiceImplTest extends Specification {
...
-				{ Date updatedAt ->
-					assert DateUtils.roughlyEqual(updatedAt, new Date())
+				{ UpdateImportRequestStatusDbDto request ->
+					assert request?.requestId == expectedRequestId
+					assert DateUtils.roughlyEqual(request?.date, new Date())
+					assert request?.oldStatus == SeriesImportRequestStatus.PARSING_SUCCEEDED 
+					assert request?.newStatus == SeriesImportRequestStatus.IMPORT_SUCCEEDED
 					return true
-				}
+				}
```
Comment (php-coder) [inline src/test/groovy/ru/mystamps/web/feature/series/importing/SeriesImportServiceImplTest.groovy]: Please, remove trailing spaces/tabs.

File: src/test/groovy/ru/mystamps/web/feature/series/importing/SeriesImportServiceImplTest.groovy [hunk 3]
```diff
@@ -258,14 +258,14 @@ class SeriesImportServiceImplTest extends Specification {
 			)
 		then:
 			1 * seriesImportDao.setSeriesIdAndChangeStatus(
-				expectedRequestId,
 				expectedSeriesId,
-				SeriesImportRequestStatus.PARSING_SUCCEEDED,
-				SeriesImportRequestStatus.IMPORT_SUCCEEDED,
-				{ Date updatedAt ->
-					assert DateUtils.roughlyEqual(updatedAt, new Date())
+				{ UpdateImportRequestStatusDbDto request ->
```
Comment (php-coder) [inline src/test/groovy/ru/mystamps/web/feature/series/importing/SeriesImportServiceImplTest.groovy]: Let's rename it to `status` (for consistency with the tests for `changeStatus()`).

File: src/main/java/ru/mystamps/web/feature/series/importing/SeriesImportServiceImpl.java [hunk 4]
```diff
@@ -98,7 +97,6 @@ public Integer addSeries(
 		AddSeriesSalesDto saleDto,
 		Integer requestId,
 		Integer userId) {
-
```
Comment (php-coder) [inline src/main/java/ru/mystamps/web/feature/series/importing/SeriesImportServiceImpl.java]: Please, don't change anything that isn't related to the issue. The commit should be focused only on a single change.

File: src/main/java/ru/mystamps/web/feature/series/importing/JdbcSeriesImportDao.java [hunk 5]
```diff
@@ -25,7 +25,6 @@
 import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
 import org.springframework.jdbc.support.GeneratedKeyHolder;
 import org.springframework.jdbc.support.KeyHolder;
-
```
Comment (php-coder) [inline src/main/java/ru/mystamps/web/feature/series/importing/JdbcSeriesImportDao.java]: I'm surprised that checkstyle doesn't complain about this change... How about reverting it?

File: src/main/java/ru/mystamps/web/feature/series/importing/SeriesImportServiceImpl.java [hunk 6]
```diff
@@ -112,13 +110,13 @@ public Integer addSeries(
 		Date now = new Date();
 		
 		seriesImportDao.setSeriesIdAndChangeStatus(
-			requestId,
 			seriesId,
-			SeriesImportRequestStatus.PARSING_SUCCEEDED,
-			SeriesImportRequestStatus.IMPORT_SUCCEEDED,
-			now
+			new UpdateImportRequestStatusDbDto(
```
Comment (php-coder) [inline src/main/java/ru/mystamps/web/feature/series/importing/SeriesImportServiceImpl.java]: It's better to follow the same approach everywhere. See how we have done the same in the `changeStatus()`:
```java
		UpdateImportRequestStatusDbDto status =
			new UpdateImportRequestStatusDbDto(requestId, now, oldStatus, newStatus);
```
Let's do the same here.

File: src/main/java/ru/mystamps/web/feature/series/importing/JdbcSeriesImportDao.java [hunk 7]
```diff
@@ -97,26 +96,21 @@ public Integer add(ImportSeriesDbDto importRequest) {
...
 	@Override
-	public void setSeriesIdAndChangeStatus(
-		Integer requestId,
-		Integer seriesId,
-		String oldStatus,
-		String newStatus,
-		Date updatedAt) {
-		
+	public void setSeriesIdAndChangeStatus(Integer seriesId,
+										   UpdateImportRequestStatusDbDto requestStatus) {
```
Comment (php-coder) [inline src/main/java/ru/mystamps/web/feature/series/importing/JdbcSeriesImportDao.java]: We don't format methods like that. See how the arguments wrapped in other places and follow the style.

Another option (if the line just a little long or you need to have everything in one like to improve readability) -- add a special comment and suppress checkstyle complains.

File: src/main/java/ru/mystamps/web/feature/series/importing/SeriesImportServiceImpl.java [hunk 8]
```diff
@@ -37,7 +37,6 @@
 import ru.mystamps.web.feature.series.sale.SeriesSalesService;
 import ru.mystamps.web.support.spring.security.HasAuthority;
 import ru.mystamps.web.util.CatalogUtils;
-
```
Comment (php-coder) [inline src/main/java/ru/mystamps/web/feature/series/importing/SeriesImportServiceImpl.java]: Please, revert this change.

File: src/main/java/ru/mystamps/web/feature/series/importing/JdbcSeriesImportDao.java (hunk 1 above)
Comment (KrivenkoAlexander) [reply src/main/java/ru/mystamps/web/feature/series/importing/JdbcSeriesImportDao.java]: done

File: src/test/groovy/ru/mystamps/web/feature/series/importing/SeriesImportServiceImplTest.groovy (hunk 2 above)
Comment (KrivenkoAlexander) [reply src/test/groovy/ru/mystamps/web/feature/series/importing/SeriesImportServiceImplTest.groovy]: done

File: src/test/groovy/ru/mystamps/web/feature/series/importing/SeriesImportServiceImplTest.groovy (hunk 3 above)
Comment (KrivenkoAlexander) [reply src/test/groovy/ru/mystamps/web/feature/series/importing/SeriesImportServiceImplTest.groovy]: done

File: src/main/java/ru/mystamps/web/feature/series/importing/SeriesImportServiceImpl.java (hunk 4 above)
Comment (KrivenkoAlexander) [reply src/main/java/ru/mystamps/web/feature/series/importing/SeriesImportServiceImpl.java]: it was reverted back . Sorry, it is my mad :(

File: src/main/java/ru/mystamps/web/feature/series/importing/JdbcSeriesImportDao.java (hunk 5 above)
Comment (KrivenkoAlexander) [reply src/main/java/ru/mystamps/web/feature/series/importing/JdbcSeriesImportDao.java]: I  am surprised too .Because i remember that i  had to change such order in previous  PRs

File: src/main/java/ru/mystamps/web/feature/series/importing/SeriesImportServiceImpl.java (hunk 6 above)
Comment (KrivenkoAlexander) [reply src/main/java/ru/mystamps/web/feature/series/importing/SeriesImportServiceImpl.java]: Done . I explicit created the status variable  and added  it to the  method call.

File: src/main/java/ru/mystamps/web/feature/series/importing/JdbcSeriesImportDao.java (hunk 7 above)
Comment (KrivenkoAlexander) [reply src/main/java/ru/mystamps/web/feature/series/importing/JdbcSeriesImportDao.java]: Done. I used suppress annotation for "linelength" module

File: src/main/java/ru/mystamps/web/feature/series/importing/SeriesImportServiceImpl.java (hunk 8 above)
Comment (KrivenkoAlexander) [reply src/main/java/ru/mystamps/web/feature/series/importing/SeriesImportServiceImpl.java]: done

**Comment by php-coder** [comment]
Merged in 86042dc57ffad891c13dc13273450d8912471fd8 commit.

**Comment by php-coder** [comment]
@KrivenkoAlexander Thank you for the contribution! 👍
